docs/source/scripts/exportdb/generate_example_rst.py

```diff
@@ -0,0 +1,184 @@
+    """
+    Generates the .rst files for all the examples in db/examples/
+    """
+
+    model = example_case.model
+
+    tags = ", ".join(f":doc:`{tag} <{tag}>`" for tag in example_case.tags)
+
+    source_file = (
+        inspect.getfile(model.__class__)
+        if isinstance(model, torch.nn.Module)
+        else inspect.getfile(model)
+    )
+    with open(source_file, "r") as file:
+        source_code = file.read()
+    source_code = re.sub(r"from torch\._export\.db\.case import .*\n", "", source_code)
+    source_code = re.sub(r"@export_case\((.|\n)*?\)\n", "", source_code)
+    source_code = source_code.replace("\n", "\n    ")
+    splitted_source_code = re.split(r"@export_rewrite_case.*\n", source_code)
+
+    assert len(splitted_source_code) in {
+        1,
+        2,
+    }, f"more than one @export_rewrite_case decorator in {source_code}"
+
+    # Generate contents of the .rst file
+    title = f"{example_case.name}"
+    doc_contents = f"""{title}
+{'^' * (len(title))}
+
+.. note::
+
+    Tags: {tags}
+
+    Support Level: {example_case.support_level.name}
+
+Original source code:
+
+.. code-block:: python
+
+    {splitted_source_code[0]}
+
+Result:
+
+.. code-block::
+
+"""
+
+    # Get resulting graph from dynamo trace
+    try:
+        inputs = normalize_inputs(example_case.example_inputs)
+        exported_program = export(
+            model,
+            inputs.args,
+            constraints=example_case.constraints,
+        )
+        graph_output = str(exported_program)
+        graph_output = re.sub(r"        # File(.|\n)*?\n", "", graph_output)
+        graph_output = graph_output.replace("\n", "\n    ")
+        output = f"    {graph_output}"
+    except torchdynamo.exc.Unsupported as e:
+        output = "    Unsupported: " + str(e).split("\n")[0]
+
+    doc_contents += output + "\n"
+
+    if len(splitted_source_code) == 2:
+        doc_contents += f"""\n
+You can rewrite the example above to something like the following:
+
+.. code-block:: python
+
+{splitted_source_code[1]}
+
+"""
+
+    # with open(os.path.join(EXPORTDB_SOURCE, f"{example_case.name}.rst"), "w") as f:
+    #     f.write(doc_contents)
+
+    return doc_contents
+
+
+def generate_index_rst(example_cases, tag_to_modules, support_level_to_modules):
+    """
+    Generates the index.rst file
+    """
+
+    # support_level_to_modules = {}
+    # for _, example_case in sorted(example_cases.items()):
+    #     support_modules = support_level_to_modules.setdefault(
+    #         example_case.support_level, []
+    #     )
+    #     support_modules.append(example_case.name)
+
+    support_contents = ""
+    for k, v in support_level_to_modules.items():
+        support_level = k.name.lower().replace("_", " ").title()
+        module_contents = "\n\n".join(v)
+        support_contents += f"""
+{support_level}
+{'-' * (len(support_level))}
+
+{module_contents}
+"""
+
+    tag_names = "\n    ".join(t for t in tag_to_modules.keys())
```

Comment:
I noticed there're "tag" appended to every tag name in the list, why is that? IMO better remove them.

Reply:
ok